machines/vic/reification/appliance.go

```diff
@@ -0,0 +1,416 @@
+				FileOperation: types.VirtualDeviceConfigSpecFileOperationCreate,
+				Operation:     types.VirtualDeviceConfigSpecOperationAdd,
+				Device: &types.VirtualDisk{
+					CapacityInKB: 16 * 1024 * 1024, //16 GB for metadata
+					VirtualDevice: types.VirtualDevice{
+						Key:           -1,
+						ControllerKey: 100,
+						UnitNumber:    &unitNumber,
+						Backing: &types.VirtualDiskFlatVer2BackingInfo{
+							DiskMode:        string(types.VirtualDiskModeIndependent_persistent),
+							ThinProvisioned: &thin,
+							VirtualDeviceFileBackingInfo: types.VirtualDeviceFileBackingInfo{
+								FileName: fmt.Sprintf("[%s]", conf.ImageDatastoreName),
+							},
+						},
+					},
+				},
+			},
+			&types.VirtualDeviceConfigSpec{
+				Operation: types.VirtualDeviceConfigSpecOperationAdd,
+				Device: &types.ParaVirtualSCSIController{
+					VirtualSCSIController: types.VirtualSCSIController{
+						SharedBus: types.VirtualSCSISharingNoSharing,
+						VirtualController: types.VirtualController{
+							BusNumber: 0,
+							VirtualDevice: types.VirtualDevice{
+								Key: 100,
+							},
+						},
+					},
+				},
+			},
+			// ide controller for cdrom
+			&types.VirtualDeviceConfigSpec{
+				Operation: types.VirtualDeviceConfigSpecOperationAdd,
+				Device: &types.VirtualIDEController{
+					VirtualController: types.VirtualController{
+						VirtualDevice: types.VirtualDevice{
+							Key: 200,
+						},
+					},
+				},
+			},
+		},
+	}
+	spec.DeviceChange = append(spec.DeviceChange, netDevices...)
+	return spec, nil
+}
+
+func (d *Dispatcher) findAppliance(conf *config.Config) (*vm.VirtualMachine, error) {
+	if ovm, err := d.session.Finder.VirtualMachine(d.ctx, conf.DisplayName); err != nil {
+		_, ok := err.(*find.NotFoundError)
+		if !ok {
+			err = errors.Errorf("Failed to query appliance (%s): %s", conf.DisplayName, err)
+			return nil, err
+		} else {
+			log.Debugf("Appliance is not found")
+			return nil, nil
+		}
+	} else {
+		newVM := vm.NewVirtualMachine(d.ctx, d.session, ovm.Reference())
+		// workaround here. We lost the value set in ovm cause we wrap the object to another type
+		newVM.InventoryPath = ovm.InventoryPath
+		return newVM, nil
+	}
+}
+
+func (d *Dispatcher) createAppliance(conf *config.Config) error {
+	log.Infof("Creating appliance on target")
+
+	spec, err := d.createApplianceSpec(conf)
+	if err != nil {
+		log.Errorf("Unable to create appliance spec: %s", err)
+		return err
+	}
+
+	// create test VM
+	info, err := tasks.WaitForResult(d.ctx, func(ctx context.Context) (tasks.ResultWaiter, error) {
+		return d.session.Folders(ctx).VmFolder.CreateVM(ctx, *spec, d.vchPool.ResourcePool, d.session.Host)
+	})
+
+	if err != nil {
+		log.Errorf("Unable to create appliance VM: %s", err)
+		return err
+	}
+	if info.Error != nil || info.State != types.TaskInfoStateSuccess {
+		log.Errorf("Create appliance reported: %s", info.Error.LocalizedMessage)
+	}
+
+	// get VM
+	vm, err := d.findAppliance(conf)
```

Comment:
I remember why I want to use this method. The vm.InventoryPath is used in the later VM configuration. And the value vm.InventoryPath is filled by govmomi Finder only.